### DISCUSSION TITLE
How to repeat texture?

### DISCUSSION THREAD

**Discussion by paulocoutinhox** [axmolengine/axmol#1454] [Q&A]
Hi,

I have a sprint that i want repeat, but not only for opengl, but in a way that work on all devices (ios, android, tvos, macos).

Some old examples:
- https://discuss.cocos2d-x.org/t/fill-sprite-with-repeating-texture/16500/4
- https://discuss.cocos2d-x.org/t/how-to-repeat-an-image-like-a-pattern/27227/6

Example of image to repeat:
- https://github.com/axmolengine/axmol/blob/dev/tests/cpp-tests/Content/Images/blocks.png

Can anyone help me?

Thanks.

**Comment by rh101** [comment]
Show the code you used in your attempt to display the repeated texture.  The samples you linked work when ported over to Axmol, so there must be something you missed in your implementation.  

Here is the code running on Axmol (from the first link in your post):
![image](https://github.com/axmolengine/axmol/assets/8603230/11c4abbf-23b7-4cb5-a97e-5df7b6741fa1)

**Reply by paulocoutinhox** [reply]
Hi,

I don't have a code that do it, because i don't know how to do.

Today im putting the images side-by-side.

I want repeat this image horizontally only:
![06](https://github.com/axmolengine/axmol/assets/395096/347ba9e7-4ee1-492d-ad74-af65428239da)


Thanks.

**Comment by rh101** [comment]
> I don't have a code that do it, because i don't know how to do.

For the first link you provided, all that needed to be done is to convert it to CPP and use the relevant texture parameters:

```
auto sprite = Sprite::create("blocks.png");
sprite->setAnchorPoint(Vec2::ANCHOR_MIDDLE);
sprite->setPosition(visibleSize.width / 2, visibleSize.height / 2);
addChild(sprite, 0);

sprite->getTexture()->setTexParameters({ backend::SamplerFilter::LINEAR, backend::SamplerFilter::LINEAR, backend::SamplerAddressMode::REPEAT, backend::SamplerAddressMode::REPEAT });
sprite->setTextureRect(Rect(0, 0, 480, 800));
```

As you can see, it's exactly the same code, just in CPP.  Also, it should work on all platforms, and not just ones that use OpenGL.

> I want repeat this image horizontally only:

Now that you have working code, I'm sure you'll figure out how to do that.
